src/monitor/group_state_machine.c

```diff
@@ -957,6 +957,35 @@ ProceedGroupStateForPrimaryNode(AutoFailoverNode *primaryNode)
 		}
 	}
 
+	/*
+	 * when another node wants to go to maintenance:
+	 *  apply_settings -> primary
+	 *  wait_maintenance -> maintenance
+	 */
+	if (IsCurrentState(primaryNode, REPLICATION_STATE_APPLY_SETTINGS))
+	{
+		ListCell *nodeCell = NULL;
+
+		foreach(nodeCell, otherNodesGroupList)
+		{
+			AutoFailoverNode *otherNode = (AutoFailoverNode *) lfirst(nodeCell);
+
+			if (IsCurrentState(otherNode, REPLICATION_STATE_WAIT_MAINTENANCE))
```

Comment:
this is currently never reached because there's the this check that will cause a transition from wait_maintenance to maintenance whenever target state of the primary isn't wait_primary:
https://github.com/citusdata/pg_auto_failover/blob/master/src/monitor/group_state_machine.c#L506-L507